Aptoide/aptoide-client-v8
# Feature/asv 1037 ab test apk fy notification

**What does this PR do?**

This PR has two different objectives:

- Add a new APKFY flow and manage it with calls to the abtesting platform;

- Hide update notifications according to a call to abtesting;

On the second point, the way AbTestManager is being created inside the class is due to the fact that we can't get the google ad id on the main thread, which makes it impossible to inject into AptoideApplication (as far as I can tell). 

**Database changed?**

 No

**Where should the reviewer start?**

- [ ] NewAppViewFragment.kt

**How should this be manually tested?**

Notifications can be tested by:

- Install app;

- Check charles to see if you're in the "hide" bucket;

- If you are, make sure you don't see the push notification for updates;

- Clear app data;

- Put a breakpoint on that request to abtesting. Once the response comes through, change the "hide" to "show";

- Make sure you see the push notification now;

P.S.: If start out in the "show" bucket, do the same steps above but change "show" to "hide" and check the behavior is correct.

APFKY can be tested by installing the app as is described in the ticket. This launches the apkfy flow. Use the same technique described above with charles to change the bucket you're assigned to and verify both flows;

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1037](<https://aptoide.atlassian.net/browse/ASV-1037>)

**Code Review Checklist**
